### ISSUE TITLE
Alignment of zero-element arrays differs

### ISSUE THREAD

**Issue by Nemo157** [fizyk20/generic-array#106]
```rust
std::mem::align_of::<[u32; 0]>() == 4;
std::mem::align_of::<generic_array::GenericArray::<u32, typenum::U0>>() == 1;
```

This is because a `GenericArray<T, U0>` only contains a `()` field, changing this base case to be a `[T; 0]` seems like it could plausibly fix it.

**Comment by novacrazy** [comment]
`[T; 0]` does seem to work, all tests pass.

**Comment by novacrazy** [comment]
Fixed in `0.14.3`
